### DISCUSSION TITLE
How can I build deep nested form with tabs structure with single submit button?

### DISCUSSION THREAD

**Discussion by ArchanaSharma95** [react-hook-form/react-hook-form#10116] [Q&A]
I am building a form which contains an array of objects of tab and then there further nested data. I'm getting stuck in the deep hierarchy where I have got or registered every value like "tabname.actions.innerObj.actions.detail...." and so on. 

I want to know what is the best method when you have built huge and deep nested forms. I have checked your tab-based example but it's coming with a different submit button. https://codesandbox.io/s/tab-form-34oio?file=/src/Form1.tsx

But in my case, I'm having multiple tabs, with different values and with the common submit button. 

**Requirements:**
- Please suggest how can I split this type of structure and further how can I get complete form data inside a single form on submission.
- Also, if splitting into small forms how will I Update the changes to the main form?

I am building this type of structure:

```
{
	"onPress": {
		"actions": {
			"fdsdsfdsfdsf": {
				"_id": "fdsdsfdsfdsf",
				"name": "A1",
				"objType": "action",
				"action": "myaction",    
				"detail": {
					"screenId": "",
					"backNav": true,
					"replace": false,
					"animation": "",
					"duration": 1000,
					"params": [],
				}
			},
			"fdsd56234sfdsf": {
				"_id": "fdsd56234sfdsf",
				"name": "A2",
				"objType": "action",
				"action": "",// detail will only come if this field is selected    
			},
		},
		roots: ["a1", "a2"],
	},
	...
}
```
Here "onPress" is a tab and further its object indicates that tab data. I can have multiple tabs. Inside the tab structure, each object inside "actions" indicate one item and it can grow to any limit. Also inside each action object, we have an " action" field if it is set only then we need to build one more structure which is its detail.

To show this I am having a main file where I am building a tab and handling each object data detail using separate child components. No problem is I have to pass field names as: "onPress.actions.{its id}.detail.backNav" to change the "backNav" value.

Please help to suggest how can I build this type of complex form with easy methods, so that it can be easy to modify or grow further.
 Any help would be appreciated, thanks!

**Comment by leapful** [comment]
Close as it's duplicated with https://github.com/react-hook-form/react-hook-form/discussions/10124
